sdk/cosmosdb/cosmos/src/auth.ts

```diff
@@ -92,6 +106,167 @@ export async function setAuthorizationTokenHeaderUsingMasterKey(
+  resourceType: ResourceType,
+  headers: CosmosHeaders,
+  masterKey: string,
+  sasTokenProperties?: SasTokenProperties
+) {
+  var resourcePrefixPath = "";
+  if (
+    typeof sasTokenProperties.databaseName === "string" &&
+    sasTokenProperties.databaseName !== ""
+  ) {
+    resourcePrefixPath += `/${Constants.Path.DatabasesPathSegment}/${sasTokenProperties.databaseName}`;
+  }
+
+  if (
+    typeof sasTokenProperties.containerName === "string" &&
+    sasTokenProperties.containerName !== ""
+  ) {
+    if (sasTokenProperties.databaseName === "") {
+      throw new Error(`illegalArgumentException : ${sasTokenProperties.databaseName} \
+                          is an invalid database name`);
+    }
+    resourcePrefixPath += `/${Constants.Path.CollectionsPathSegment}/${sasTokenProperties.containerName}`;
+  }
+
+  if (
+    typeof sasTokenProperties.resourceName === "string" &&
+    sasTokenProperties.resourceName !== ""
+  ) {
+    if (sasTokenProperties.containerName === "") {
+      throw new Error(`illegalArgumentException : ${sasTokenProperties.containerName} \
+                          is an invalid container name`);
+    }
+    switch (sasTokenProperties.resourceKind) {
+      case "ITEM":
+        resourcePrefixPath += `${Constants.Path.Root}${Constants.Path.DocumentsPathSegment}`;
+        break;
+      case "STORED_PROCEDURE":
+        resourcePrefixPath += `${Constants.Path.Root}${Constants.Path.StoredProceduresPathSegment}`;
+        break;
+      case "USER_DEFINED_FUNCTION":
+        resourcePrefixPath += `${Constants.Path.Root}${Constants.Path.UserDefinedFunctionsPathSegment}`;
+        break;
+      case "TRIGGER":
+        resourcePrefixPath += `${Constants.Path.Root}${Constants.Path.TriggersPathSegment}`;
+        break;
+      default:
+        throw new Error(`illegalArgumentException : ${sasTokenProperties.resourceKind} \
+                          is an invalid resource kind`);
+        break;
+    }
+    resourcePrefixPath += `${Constants.Path.Root}${sasTokenProperties.resourceName}${Constants.Path.Root}`;
+  }
+
+  sasTokenProperties.resourcePath = resourcePrefixPath.toString();
+  var partionRanges = "";
+
+  if (
+    sasTokenProperties.partitionKeyValueRanges !== undefined &&
+    sasTokenProperties.partitionKeyValueRanges.length > 0
+  ) {
+    if (
+      typeof sasTokenProperties.resourceKind !== "string" &&
+      sasTokenProperties.resourceKind !== "ITEM"
+    ) {
+      throw new Error(`illegalArgumentException : ${sasTokenProperties.resourceKind} \
+                          is an invalid partition key value range`);
+    }
+    sasTokenProperties.partitionKeyValueRanges.forEach((range) => {
+      partionRanges += `${encodeUTF8(range)},`;
+    });
+  }
+
+  if (sasTokenProperties.controlPlaneReaderScope === 0) {
+    sasTokenProperties.controlPlaneReaderScope += SasTokenPermissionKind.ContainerReadAny;
+    sasTokenProperties.controlPlaneWriterScope += SasTokenPermissionKind.ContainerReadAny;
```

Comment:
is the writer scope supposed to be updated if the reader scope is 0?

Reply:
Experimental: this logic will grow with the feature.